src/pages/Home.jsx

```diff
@@ -1,41 +1,58 @@
 import { useEffect, useState } from "react";
 import SingleBook from "../components/SingleBook";
 import { Link } from "react-router-dom";
+import GenreFilter from "../components/GenreFilter";
 
 export default function Home({claimed}) {
     const [bookInfo, setBookInfo] = useState([])
+    const [filteredBooks, setFilteredBooks] = useState([])
+    const [genres, setGenres] = useState([])
+    const [selectedGenre, setSelectedGenre] = useState("")
 
     function getData() {
         const url = `https://book-swap-api.dev.io-academy.uk/api/books?claimed=${claimed}`
-        // if ({claimed}=1){
-        //    url = "https://book-swap-api.dev.io-academy.uk/api/books?claimed=1"
-        // }
-        // else {
-        //     url = "https://book-swap-api.dev.io-academy.uk/api/books?claimed=0"
-        // }
 
         fetch(url)
          .then(res => res.json())
          .then(fetchedInfo => {
-            setBookInfo(fetchedInfo.data)
+            setBookInfo(fetchedInfo.data);
+            setFilteredBooks(fetchedInfo.data) 
+
+            const uniqueGenres = Array.from (new Set(fetchedInfo.data.map(book => book.genre.name)));
```

Comment:
Ok so this is a cool solution to finding the genres, however, you should instead be using the API for this. It does have a route that can give you the list of genres.

If it didn't, this solution would be perfect.